packages/components/src/components/primitives/Link/types.ts

```diff
@@ -0,0 +1,13 @@
+import { LinkProps as ChakraLinkProps } from '@chakra-ui/react';
+
+export const linkIconPositions = ['leading', 'trailing'] as const;
+export const linkSizes = ['s', 'm', 'l'] as const;
+
+type LinkIconPositions = typeof linkIconPositions[number];
+type LinkSizes = typeof linkSizes[number];
+
+export interface LinkProps extends ChakraLinkProps {
+  icon?: JSX.Element;
+  iconPosition?: LinkIconPositions;
+  size?: LinkSizes | string;
```

Comment:
Should a `Link` accept a different `size` that the ones we've defined? Maybe we can remove the `| string`